platform/android/export/export_plugin.cpp

```diff
@@ -2382,6 +2387,16 @@ bool EditorExportPlatformAndroid::has_valid_project_configuration(const Ref<Edit
 		err += "\n";
 	}
 
+	//Check package name when used $genname in the "Unique Name" field
+	String basename = GLOBAL_GET("application/config/name");
+	String name = get_valid_basename();
+	if (valid && pn.find("$genname") >= 0 && name != basename) {
+		String p_name = p_preset->get("package/unique_name");
+		String fn = get_package_name(p_name);  //The final package name
+		err += vformat(TTR("Invalid package name. Will be replaced with \"%s\" when exporting."), fn);
+		err += "\n";
+	}
+
```

Comment:
There is a check above that already validates the package name, can this logic be added to it.

Reply:
@m4gr3d if I do the check above, then the variable `valid` may then change to the `false`. Then this warning will be displayed in red font, which will be wrong. If the export is not possible, then this warning is not necessary.
Here I need to delete line number 2394 and replace `p_name` with `pn`.
This is my first pull request and I don't now how to do it yet, because I messed up something with the branches. I will try to figure it out.

Reply:
I think the warning should be turned into an error, in which case it's fine to merge the added logic with `is_package_name_valid`.

The reason for being an error is that a warning that doesn't block the user from progressing can easily be overlooked leading to confusion when the package name doesn't match with the unique name.
I'd rather the export dialog to be explicit to the user that the selected name is invalid.

Reply:
As part of the error, we can suggest the user to manually overwrite the package name with a suitable replacement if they would like to keep the unique name.